### DISCUSSION TITLE
Deploying to a selected domain using REST API.

### DISCUSSION THREAD

**Discussion by ritvij14** [vercel/community#5984] [Feature Question]
### Question

Our team has a particular use case using the REST API where we have to allow users to preview the deployed website before putting it out on the main domain.

We are gonna be using GitHub APIs to push code to their project which will be connected to vercel to deploy the things. Is it possible that when the user wants to preview the deployed sit, we can deploy the code to a preview URL first and then when the user selects publish, only then it gets deployed to the main domain?

To summarise:
- User edits their website (ours is a template website builder like Square space)
- we take the edits, convert them into code, and push that into the github project (which is connected to Vercel)
- when user chooses to preview the new data is deployed only to a preview link on vercel
- only when user selects deploy, then we ask Vercel using the REST API to create a final deployment to the main domain.
